### PR TITLE
Move proxy_main.py to be included in codejail module

### PR DESCRIPTION
**Pull request by ariellasmo**
Move `proxy_main.py` to be installable as a library.
[MLA-588](https://korbit-ai.atlassian.net/browse/MLA-588)

### CODE REVIEW

**Comment by ariellasmo** [comment]
This will be a breaking change on the default sandbox. Let's not merge until default sandbox uses codejail properly as a proper library.

File: codejail/proxy.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def get_proxy():
     # If we need a proxy, make a proxy.
     if PROXY_PROCESS is None:
         # Start the proxy by invoking proxy_main.py in our root directory.
-        root = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+        root = os.path.dirname(os.path.realpath(__file__))
```
Comment (antoineKorbit) [inline codejail/proxy.py:116]: How this is a breaking change we aren't using PROXY process in codippy default sandbox 🤔

Comment (ariellasmo) [reply codejail/proxy.py:116]: I haven't actually looked at the codippy default sandbox at all. I didn't know `proxy_main.py` was only used in the codippy validation sandbox.

Do you think this PR can be merged now @antoineKorbit ?

Comment (antoineKorbit) [reply codejail/proxy.py:116]: Oh I spotted it, it's declared in the Docker container as ENV variable.

**Pull request merged**